File: Java/program2/Main.java
package program2;
import java.util.*;

public class Main {
    public static int countRounds(int teams) {
        if (teams == 1) {
            return 0;
        }
        else if (teams % 2 == 0) {
            int matches = teams / 2;
            return matches + countRounds(matches);
        }
        else {
            int matches = (teams - 1) / 2;
            return matches + countRounds(matches+1);
        }
    }

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int totalTeams = scan.nextInt();
        scan.close();

        System.out.println(countRounds(totalTeams));
    }
}

File: Rust/program2/src/main.rs
use std::io;

fn count_rounds(teams: i32) -> i32 {
    if teams == 1 {
        0
    } else if teams % 2 == 0 {
        let matches = teams / 2;
        matches + count_rounds(matches)
    } else {
        let matches = (teams - 1) / 2;
        matches + count_rounds(matches+1)
    }
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("");
    let total_teams: i32 = input.trim().parse().expect("");
    println!("{}", count_rounds(total_teams));
}
